arduino_web_server.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# credit: http://sheep.art.pl/Wiki%20Engine%20in%20Python%20from%20Scratch
+
+
+import BaseHTTPServer
+import SimpleHTTPServer
+import platform
+import os
+import re
+import subprocess
+import tempfile
+import urllib
+from optparse import OptionParser
+
+
+if platform.system() == "Darwin":
+    arduino_cmd = "/Applications/Arduino.app/Contents/MacOS/Arduino"
+else:
+    arduino_cmd = "arduino"
+
+
+parser = OptionParser()
+parser.add_option("--port", dest="port", help="Upload to serial port named PORT", metavar="PORT")
+parser.add_option("--board", dest="board", help="Board definition to use", metavar="BOARD")
+parser.add_option("--command", dest="cmd", help="Arduino command name", metavar="CMD", default=arduino_cmd)
+
+
+class Handler(SimpleHTTPServer.SimpleHTTPRequestHandler):
+    template = u"""<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN"
+"http://www.w3.org/TR/html4/strict.dtd"><html><body><h1>Arduino INO web server</h1>To upload to an Arduino board connected to this computer, POST to /.</body></html>"""
```

Comment:
`web server to compile and upload sketch with official Arduino IDE CLI`
might be more accurate?